primitives/src/block.rs

```diff
@@ -207,6 +208,37 @@ impl Header {
     }
 }
 
+// If you have `alloc` then we can maintain the formatter options.
```

Comment:
I'm confused why alloc is needed at all if we know block header can fin in 160 hex chars.